pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -164,16 +164,35 @@ func (m *manager) calculateAffinity(pod v1.Pod, container v1.Container) Topology
 		// Get the TopologyHints from a provider.
 		hints := provider.GetTopologyHints(pod, container)
 
-		// If hints is nil, overwrite 'hints' with a preferred any-socket affinity.
+		// If hints is nil, insert a single, preferred any-socket hint into allProviderHints.
 		if hints == nil || len(hints) == 0 {
-			klog.Infof("[topologymanager] Hint Provider has no preference for socket affinity")
+			klog.Infof("[topologymanager] Hint Provider has no preference for socket affinity with any resource")
 			affinity, _ := socketmask.NewSocketMask()
 			affinity.Fill()
-			hints = []TopologyHint{{affinity, true}}
+			allProviderHints = append(allProviderHints, []TopologyHint{{affinity, true}})
+			continue
```

Comment:
Can you explain why we need Line 168-173, I was think that the loop in Line 173 should handle this. I think in any case we should look on this as hints per resource and not hints per hint Provider.